elytron/WFCORE-5744-ssl-certificate-expiration-warning.adoc

```diff
@@ -0,0 +1,218 @@
+No.
+
+== Implementation Plan
+
+N/A
+
+== Admin Clients
+
+CLI should not be affected. However, HAL will most likely require follow up, since this feature will introduce new model params/context. Given defaults, its not a breaking change.
+
+== Security Considerations
+
+Slightly increase security risk compared to current state. If attacker has access to log/events it will provide view of current health of server, rather than possible peek on startup. Convenient, but not something that could not be circumvented without this feature.
+
+[[test_plan]]
+== Test Plan
+
+Integration tests should be fairly acceptable.
+
+=== Manual Test
+
+==== Defined keystore
+* Generate certificates:
+
+[quote, shell]
+----
+keytool -genkeypair -alias localhost -keyalg RSA -keysize 2048 -validity 1 -keystore server.keystore -dname "cn=Server Administrator,o=Acme,c=GB" -keypass secret -storepass secret
+keytool -genkeypair -alias drone-1-1 -keyalg RSA -keysize 2048 -validity 365 -keystore server.keystore -dname "cn=Server Administrator,o=Acme,c=GB" -keypass secret -storepass secret
+keytool -v -list -keystore server.keystore
+keytool -importkeystore -srckeystore server.keystore -destkeystore server.keystore -deststoretype pkcs12
+cp server.keystore ${SRV_HOME}/standalone/configuration/
+----
+
+* Enable keystore:
+
+[quote, JBoss CLI]
+----
+/subsystem=elytron/key-store=demoKeyStore:add(path=server.keystore,relative-to=jboss.server.config.dir, credential-reference={clear-text=secret},type=JKS)
+/subsystem=elytron/key-manager=demoKeyManager:add(key-store=demoKeyStore,credential-reference={clear-text=secret})
+/subsystem=elytron/server-ssl-context=demoSSLContext:add(key-manager=demoKeyManager,protocols=["TLSv1.2"])
+/subsystem=undertow/server=default-server/https-listener=https:write-attribute(name=ssl-context,value=demoSSLContext)
+:reload
+----
+
+===== Periodic
+
+* Keystore config:
+
+[quote, JBoss CLI]
+----
+/subsystem=elytron/key-store=demoKeyManager:write-attribute(name=expiration-check-delay, value=1000)
+:reload
+----
+
+
+===== Read alias
+
+[quote, JBoss CLI]
+----
+/subsystem=elytron/key-store=demoKeyStore:read-alias(alias=localhost)
+----
+
+[quote, JBoss CLI]
+----
+{
+    "outcome" => "success",
+    "result" => {
+        "alias" => "localhost",
+        "entry-type" => "PrivateKeyEntry",
+        "creation-date" => "2025-02-11T15:25:40.316+0100",
+        "certificate-chain" => [{
+            "type" => "X.509",
+            "algorithm" => "RSA",
+            "format" => "X.509",
+            "public-key" => "...",
+            ...
+            "not-before" => "2025-02-11T15:22:47.000+0100",
+            "not-after" => "2025-02-12T15:22:47.000+0100",
+            ...
+            "validity" => "EXPIRED"
+        }]
+    }
+}
+----
+
+* Global config:
+
+== Community Documentation
+
+Model/XSD description should be enough as change is not deep and only introudce simple config parameter and additional output KVP in existing command.
```

Comment:
I think there should be a little more in the community documentation, if a user is activating (assuming it is them activating it) a periodic check I think we should have a small paragraph to let them know that that check entails.